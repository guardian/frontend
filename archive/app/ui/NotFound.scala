package ui

import rendering.Renderable
import play.api.libs.json.JsValue

object NotFound extends Renderable {
  override def props: Option[JsValue] = None
}
