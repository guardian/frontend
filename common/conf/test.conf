akka {
  blocking-operations {
    executor = "thread-pool-executor"
    throughput = 10
    thread-pool-executor {
      fixed-pool-size = 128
    }
  }
}
