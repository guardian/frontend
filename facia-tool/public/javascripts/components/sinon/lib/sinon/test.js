/**
 * @depend ../sinon.js
 * @depend stub.js
 * @depend mock.js
 * @depend sandbox.js
 */
/*jslint eqeqeq: false, onevar: false, forin: true, plusplus: false*/
/*global module, require, sinon*/
/**
 * Test function, sandboxes fakes
 *
 * @author Christian Johansen (christian@cjohansen.no)
 * @license BSD
 *
 * Copyright (c) 2010-2013 Christian Johansen
 */
"use strict";

(function (sinon) {
    var commonJSModule = typeof module == "object" && typeof require == "function";

    if (!sinon && commonJSModule) {
        sinon = require("../sinon");
    }

    if (!sinon) {
        return;
    }

    function test(callback) {
        var type = typeof callback;

        if (type != "function") {
            throw new TypeError("sinon.test needs to wrap a test function, got " + type);
        }

        return function () {
            var config = sinon.getConfig(sinon.config);
            config.injectInto = config.injectIntoThis && this || config.injectInto;
            var sandbox = sinon.sandbox.create(config);
            var exception, result;
            var args = Array.prototype.slice.call(arguments).concat(sandbox.args);

            try {
                result = callback.apply(this, args);
            } catch (e) {
                exception = e;
            }

            if (typeof exception !== "undefined") {
                sandbox.restore();
                throw exception;
            }
            else {
                sandbox.verifyAndRestore();
            }

            return result;
        };
    }

    test.config = {
        injectIntoThis: true,
        injectInto: null,
        properties: ["spy", "stub", "mock", "clock", "server", "requests"],
        useFakeTimers: true,
        useFakeServer: true
    };

    if (commonJSModule) {
        module.exports = test;
    } else {
        sinon.test = test;
    }
}(typeof sinon == "object" && sinon || null));
