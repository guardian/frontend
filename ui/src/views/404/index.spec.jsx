// @flow

import test from 'utils/test-render';
import NotFound from './index';

jest.mock('./style.css', () => ({
    message: {
        color: 'hotpink',
    },
}));

describe('404', () => {
    const config = {
        beaconUrl: '',
    };
    test(<NotFound config={config} />);
});
