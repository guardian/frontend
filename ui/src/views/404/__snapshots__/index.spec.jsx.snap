// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 generates the expected markup 1`] = `
Object {
  "css": "<style class=\\"_styletron_hydrate_\\"></style>",
  "html": "<div><svg><text>test svg</text></svg><h1>Sorry - we havenâ€™t been able to serve the page you asked for.</h1><p>You may have followed a broken or outdated link, or there may be an error on our site.</p><p>Please follow one of the links below to continue exploring.</p></div>",
}
`;
